go-controller/pkg/ovn/pods.go

```diff
@@ -390,6 +417,33 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) (err error) {
 		}
 	}
 
+	// add src-ip routes to GR if external gw annotation is set
+	routingExternalGWs, err := oc.getRoutingExternalGWs(pod.Namespace)
+	if err != nil {
+		return err
+	}
```

Comment:
so in add logical port, if snat is disabled, you are going to need to check if routingExternalGws is false on this namespace. If it is false, then you need to configure an snat on the GR only for this pod's IP. It must also be cleaned up on pod deletion.

Reply:
ack

Reply:
done